phfs/phoenixd.c

```diff
@@ -148,8 +161,34 @@ ssize_t phoenixd_read(unsigned int fd, unsigned int major, unsigned int minor, a
 
 ssize_t phoenixd_write(unsigned int fd, unsigned int major, unsigned int minor, addr_t offs, const void *buff, size_t len)
 {
-	/* TODO */
-	return EOK;
+	msg_t smsg, rmsg;
+	msg_phoenixd_t *io;
+	u16 hdrsz;
+
+	io = (msg_phoenixd_t *)smsg.data;
+	hdrsz = (u16)((u32)io->data - (u32)io);
```

Comment:
You assume that address is 32 bit wide. Using `&` is somewhat safer (should `io->data` become a pointer in the future for some reason).

```suggestion
	hdrsz = (u16)((addr_t)(&io->data) - (addr_t)io);
```